src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -2713,6 +2713,197 @@ void M(System.Collections.Generic.IEnumerable<int> e, int result)
+		return ValueTask.FromResult(x++ < 5);
+	}
+
+	ValueTask<bool> IAsyncEnumerator<int>.MoveNextAsync()
+	{
+		Console.WriteLine(""IAsyncEnumerator<int>.MoveNextAsync()"");
+		return MoveNextAsync();
+	}
+	ValueTask IAsyncDisposable.DisposeAsync()
+	{
+		Console.WriteLine(""IAsyncDisposable.DisposeAsync()"");
+		return DisposeAsync();
+	}
+
+	public int Current => x;
+	
+	public ValueTask DisposeAsync([CallerMemberName] string s = default,
+						[CallerLineNumber] int line = default, int xxx=12, string f = """")
+	{
+		Console.WriteLine($""line: {line}"");
+		Console.WriteLine($""member: {s}"");
+		return ValueTask.CompletedTask;
+	}
+}
+";
+
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net60);
+            comp.VerifyDiagnostics();
+            var op = (Operations.ForEachLoopOperation)VerifyOperationTreeForTest<ForEachStatementSyntax>(comp, @"IForEachLoopOperation (LoopKind.ForEach, IsAsynchronous, Continue Label Id: 0, Exit Label Id: 1) (OperationKind.Loop, Type: null) (Syntax: 'await forea ... }')
+  Locals: Local_1: System.Int32 x
+  LoopControlVariable:
+    IVariableDeclaratorOperation (Symbol: System.Int32 x) (OperationKind.VariableDeclarator, Type: null) (Syntax: 'var')
+      Initializer:
+        null
+  Collection:
+    IConversionOperation (TryCast: False, Unchecked) (OperationKind.Conversion, Type: CustomAsyncEnumerable, IsImplicit) (Syntax: 'new CustomA ... numerable()')
+      Conversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+      Operand:
+        IObjectCreationOperation (Constructor: CustomAsyncEnumerable..ctor()) (OperationKind.ObjectCreation, Type: CustomAsyncEnumerable) (Syntax: 'new CustomA ... numerable()')
+          Arguments(0)
+          Initializer:
+            null
+  Body:
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+      IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'Console.WriteLine(x);')
+        Expression:
+          IInvocationOperation (void System.Console.WriteLine(System.Int32 value)) (OperationKind.Invocation, Type: System.Void) (Syntax: 'Console.WriteLine(x)')
+            Instance Receiver:
+              null
+            Arguments(1):
+                IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: value) (OperationKind.Argument, Type: null) (Syntax: 'x')
+                  ILocalReferenceOperation: x (OperationKind.LocalReference, Type: System.Int32) (Syntax: 'x')
+                  InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                  OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+  NextVariables(0)");
+
+            //todo: replace with public api access
```

Comment:
```suggestion
```

TODO not necessary: this entire file will need to be updated with new tests/updates to existing tests if an api is added.